src/status_im/contexts/onboarding/events.cljs

```diff
@@ -136,42 +136,41 @@
-                               (re-frame/dispatch
-                                [:profile/profile-selected key-uid]))
-        :on-cancel           #(re-frame/dispatch [:pop-to-root :multiaccounts])}}
-      {:db (-> db
-               (assoc-in [:onboarding/profile :seed-phrase] seed-phrase)
-               (assoc-in [:onboarding/profile :key-uid] key-uid)
-               (assoc-in [:onboarding/profile :color] constants/profile-default-color))
-       :fx [[:dispatch
-             [:navigate-to-within-stack
-              [next-screen
-               (get db
-                    :onboarding/navigated-to-enter-seed-phrase-from-screen
-                    :screen/onboarding.new-to-status)]]]]})))
-
-(rf/defn navigate-to-create-profile
-  {:events [:onboarding/navigate-to-create-profile]}
-  [{:keys [db]}]
-  ;; Restart the flow
-  {:db       (dissoc db :onboarding/profile)
-   :dispatch [:navigate-to-within-stack
-              [:screen/onboarding.create-profile :screen/onboarding.new-to-status]]})
+(rf/reg-event-fx
+ :onboarding/seed-phrase-validated
+ (fn [{:keys [db]} [seed-phrase key-uid]]
+   (let [syncing-account-recovered? (and (seq (:syncing/key-uid db))
+                                         (= (:syncing/key-uid db) key-uid))
+         next-screen                (if syncing-account-recovered?
+                                      :screen/onboarding.create-profile-password
+                                      :screen/onboarding.create-profile)
+         from-screen                (get db
+                                         :onboarding/navigated-to-enter-seed-phrase-from-screen
+                                         :screen/onboarding.new-to-status)]
+     (if (contains? (:profile/profiles-overview db) key-uid)
+       {:fx [[:effects.utils/show-confirmation
+              {:title               (i18n/label :t/multiaccount-exists-title)
+               :content             (i18n/label :t/multiaccount-exists-content)
+               :confirm-button-text (i18n/label :t/unlock)
+               :on-accept           (fn []
+                                      (re-frame/dispatch [:pop-to-root :screen/profile.profiles])
+                                      (re-frame/dispatch [:profile/profile-selected key-uid]))
+               :on-cancel           #(re-frame/dispatch [:pop-to-root :multiaccounts])}]]}
+       {:db (-> db
+                (assoc-in [:onboarding/profile :seed-phrase] seed-phrase)
+                (assoc-in [:onboarding/profile :key-uid] key-uid)
+                (assoc-in [:onboarding/profile :color] constants/profile-default-color))
+        :fx [[:dispatch [:navigate-to-within-stack [next-screen from-screen]]]]}))))
+
+(rf/reg-event-fx
+ :onboarding/navigate-to-create-profile
+ (fn [{:keys [db]}]
+   {:db (-> db
+            (assoc-in [:onboarding/profile :color] (rand-nth colors/account-colors))
+            (assoc-in [:onboarding/profile :image-path] nil))
```

Comment:
I think maintaining a consistent usage of `contains?` is more error-prone so I'd prefer not relying on it.

`contains?` check for the existence of a key, and IMO, most of the time we want to know what is the value of the key, not if the key is present.

But I get your point, I'll use `dissoc`, I prefer it too, in my case because we keep the db smaller by continuously using it.